src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organization_policy_campaign.py

```diff
@@ -0,0 +1,552 @@
+    path: str
+    type: str
+    id: str
+    config: dict()
+
+    def __repr__(self) -> str:
+        return f"{self.path} ({self.id}) ({self.type})"
+
+    def __init__(
+        self, target_path, policy_type, policy_id, config, organizations_client=None
+    ) -> None:
+
+        self.path = target_path
+        self.type = policy_type
+        self.id = policy_id
+        self.config = config
+        self.organizations_client = (
+            organizations_client
+            if organizations_client
+            else boto3.client("organizations")
+        )
+        self.existing_policy_ids = self.get_existing_policies()
+
+    def get_existing_policies(self):
+        existing_policy_ids = {
+            p["Id"]: p["Name"]
+            for p in self.organizations_client.list_policies_for_target(
+                TargetId=self.id, Filter=self.type
+            ).get("Policies")
+        }
+        return existing_policy_ids
+
+    def attach_policy(self, policy_id, policy_name):
+        LOGGER.debug("Existing Policy Ids: %s", self.existing_policy_ids)
+        if policy_id not in self.existing_policy_ids:
+            self.organizations_client.attach_policy(
+                PolicyId=policy_id, TargetId=self.id
+            )
+            self.existing_policy_ids[policy_id] = policy_name
+        else:
+            LOGGER.info(
+                f"Policy {policy_name} ({policy_id}) already attached to {self}"
+            )
+
+        if (
+            DEFAULT_POLICY_ID in self.existing_policy_ids.keys()
+            and self.config.get("keep-default-scp", "enabled") == "disabled"
+        ):
+            self.organizations_client.detach_policy(
+                PolicyId=DEFAULT_POLICY_ID, TargetId=self.id
+            )
+
+
+class OrganizationalPolicyCampaignPolicy:
+    name: str
+    body: str
+    id: str
+    type: str
+    campaign_config: dict()
+    current_targets: list()
+    targets_requiring_attachment: dict()
+    policy_has_changed: bool
+    targets_not_scheduled_for_deletion: list()
+
+    def __repr__(self) -> str:
+        return f"{self.name} ({self.id}) ({self.type})"
+
+    def __init__(
+        self,
+        policy_name,
+        policy_body,
+        policy_type,
+        config,
+        policy_id=None,
+        policy_has_changed=False,
+        organizations_client=None,
+    ):
+        self.name = policy_name
+        self.body = policy_body
+        self.id = policy_id
+        self.type = policy_type
+        self.campaign_config = config
+        self.organizations_client = (
+            organizations_client
+            if organizations_client
+            else boto3.client("organizations")
+        )
+        self.current_targets = self.get_current_targets_for_policy()
+        self.targets_requiring_attachment = {}
+        self.policy_has_changed = policy_has_changed
+        self.targets_not_scheduled_for_deletion = []
+
+    def get_current_targets_for_policy(self):
+        if self.id:
```

Comment:
Let's flip this one, so we get rid of the indentation and the else statement:

```suggestion
        if not self.id:
          return []
```